### PR TITLE
convert script excecutor to use args array

### PR DESCRIPTION
**Pull request by krwq**

### CODE REVIEW

File: src/Microsoft.DotNet.Cli.Utils/ScriptExecutor.cs [hunk 1]
```diff
@@ -76,19 +70,15 @@ public static Command CreateCommandForScript(Project project, string scriptComma
...
                 // Unlike Windows, must escape quotation marks within the newly-quoted string.
-                scriptArguments = new[] { "/usr/bin/env", "bash", "-c", "\"" }
-                    .Concat(scriptArguments.Select(argument => argument.Replace("\"", "\\\"")))
-                    .Concat(new[] { "\"" })
-                    .ToArray();
+                scriptArguments = new[] { "/usr/bin/env", "bash", "-c", ArgumentEscaper.EscapeArgArray(scriptArguments) };
             }
 
-            return Command.Create(scriptArguments.FirstOrDefault(), string.Join(" ", scriptArguments.Skip(1)))
+            return Command.Create(scriptArguments.FirstOrDefault(), scriptArguments.Skip(1).ToArray())
```
Comment (brthor) [inline src/Microsoft.DotNet.Cli.Utils/ScriptExecutor.cs:82]: This script is somewhat of a special case, you should call this without ToArray and make above mentioned changes.

File: src/Microsoft.DotNet.Cli.Utils/ScriptExecutor.cs [hunk 2]
```diff
@@ -45,24 +45,18 @@ public static Command CreateCommandForScript(Project project, string scriptComma
                 var comSpec = Environment.GetEnvironmentVariable("ComSpec");
                 if (!string.IsNullOrEmpty(comSpec))
                 {
-                    scriptArguments =
-                        new[] { comSpec, "/S", "/C", "\"" }
-                        .Concat(scriptArguments)
-                        .Concat(new[] { "\"" })
-                        .ToArray();
+                    scriptArguments = new[] { comSpec, "/S", "/C", ArgumentEscaper.EscapeArgArray(scriptArguments) };
```
Comment (brthor) [inline src/Microsoft.DotNet.Cli.Utils/ScriptExecutor.cs]: Here you should call this EscapeArgArray Separately, then call EscapeArgArrayForCmd on that returned string

File: src/Microsoft.DotNet.Cli.Utils/ScriptExecutor.cs (hunk 1 above)
Comment (krwq) [inline src/Microsoft.DotNet.Cli.Utils/ScriptExecutor.cs:82]: I'm leaving this for now as this is too error prone to change for stabilization, we can file an issue

Comment (brthor) [inline src/Microsoft.DotNet.Cli.Utils/ScriptExecutor.cs:82]: Should not merge?

Comment (brthor) [inline src/Microsoft.DotNet.Cli.Utils/ScriptExecutor.cs:82]: If you don't change the ToArray, the args will get re-escaped and will certainly break

**Pull request merged**